### PR TITLE
feat: add sqruff as a VSCode extension

### PR DESCRIPTION
**Pull request by gvozdvmozgu**

### CODE REVIEW

File: editors/code/package.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  "main": "./dist/native.js",
+  "browser": "./dist/browser.js",
+  "contributes": {
+    "commands": [
+      {
+        "command": "code.helloWorld",
+        "title": "Hello World"
+      }
+    ]
+  },
```
Comment (benfdking) [inline editors/code/package.json]: nit: This command can be removed. 

```suggestion
  "contributes": {
    "commands": [
      {
        "command": "code.helloWorld",
        "title": "Hello World"
      }
    ]
  },
```

File: editors/code/src/browser.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const worker = new Worker(serverMain.toString(true));
+    worker.onmessage = (message) => {
+        if (message.data !== "OK") {
+            return;
+        }
+
+        const cl = new LanguageClient("sqruff-lsp", "Sqruff LSP", { documentSelector: [{ language: "sql" }] }, worker);
+        cl.start().then(() => { });
+    };
+}
```
Comment (benfdking) [inline editors/code/src/browser.ts:19]: I am struggling to see how you are testing this in the browser. Did you run `npm run run-in-browser`? and what before?

File: editors/code/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+			"ES2022",
+			"WebWorker"
+		],
+		"sourceMap": true,
+		"rootDir": "src",
+		"strict": true /* enable all strict type-checking options */
+		/* Additional Checks */
+		// "noImplicitReturns": true, /* Report error when not all code paths in function return a value. */
+		// "noFallthroughCasesInSwitch": true, /* Report errors for fallthrough cases in switch statement. */
+		// "noUnusedParameters": true,  /* Report errors on unused parameters. */
```
Comment (benfdking) [inline editors/code/tsconfig.json]: nit: I think we can turn these on.

```suggestion
		"noImplicitReturns": true, /* Report error when not all code paths in function return a value. */
		"noFallthroughCasesInSwitch": true, /* Report errors for fallthrough cases in switch statement. */
		"noUnusedParameters": true,  /* Report errors on unused parameters. */
```

File: editors/code/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+	"compilerOptions": {
+		"module": "Node16",
```
Comment (benfdking) [inline editors/code/tsconfig.json:3]: nit: This here feels slightly odd version but not important.

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -9,5 +9,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: Swatinem/rust-cache@v2
       - uses: dtolnay/rust-toolchain@nightly
-      - run: cargo r -F codegen-docs >> options.md
+      - run: cargo r -F codegen-docs > options.md
       - uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          push_options: '--force'
```
Comment (benfdking) [inline .github/workflows/docs.yml:15]: nit: we should add some CI checks for this as well.

**Review by benfdking — COMMENTED** [review]
Hey @gvozdvmozgu, 

Great work, I am just wondering, I am struggling to get it to work in browser, what are the steps that you have run to test/run it?

File: editors/code/src/browser.ts (hunk 2 above)
Comment (nikolaevx) [reply editors/code/src/browser.ts:19]: `npm run build:wasm_lsp && npm run compile && npm run run-in-browser`

**Comment by benfdking** [comment]
I think there are a few things we should improve, but in the meantime, we should merge this: 

- [ ] CI System to run and check this
  - [ ] Build and lint
- [ ] CI System that deploys this
- [ ] I do think we should register a formatter which this doesn't yet accomplish
- [ ] Document well how to run this locally

**Pull request merged**